inginious/frontend/plugins/analytics/static/calendar_view.js

```diff
@@ -0,0 +1,169 @@
+
+
+var width = 900,
+height = 105,
+cellSize = 12; // cell size
+week_days = ['Sun','Mon','Tue','Wed','Thu','Fri','Sat']
+month = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
+    
+var day = d3.time.format("%w"),
+    week = d3.time.format("%U"),
+    percent = d3.format(".1%"),
+    format = d3.time.format("%Y%m%d");
+    parseDate = d3.time.format("%Y%m%d").parse;
+        
+var no_se = ["white",  '#002b53'];
+var color = d3.scale.linear().range(['grey', 'yellow', 'green'])
+    .domain([0, 0.5,  1])
+    
+let start_year = (new Date).getFullYear()  - Number(document.getElementById('duration_time').value);
+var svg = d3.select(".calender-map").selectAll("svg")
+    .data(d3.range((new Date).getFullYear(), start_year , -1))
+.enter().append("svg")
+    .attr("width", '100%')
+    .attr("data-height", '0.5678')
+    .attr("viewBox",'0 0 900 105')
+    .attr("class", "RdYlGn")
+.append("g")
+    .attr("transform", "translate(" + ((width - cellSize * 53) / 2) + "," + (height - cellSize * 7 - 1) + ")");
+
+
+svg.append("text")
+    .attr("transform", "translate(-38," + cellSize * 3.5 + ")rotate(-90)")
+    .style("text-anchor", "middle")
+    .text(function(d) { return d; });
+
+for (var i=0; i<7; i++)
```

Comment:
What does the 7 mean? Can you have a constant previous to the for, with this value and a more understandable name?